python/cudf/cudf/_libxx/parquet.pyx

```diff
@@ -101,61 +93,50 @@ cpdef read_parquet(filepath_or_buffer, columns=None, row_group=None,
     cudf.io.parquet.to_parquet
     """
 
-    # Setup reader options
-    cdef parquet_reader_options options = parquet_reader_options()
-    for col in columns or []:
-        options.columns.push_back(str(col).encode())
-    options.strings_to_categorical = strings_to_categorical
-    options.use_pandas_metadata = use_pandas_metadata
-
-    # Create reader from source
-    cdef const unsigned char[::1] buffer = lib.view_of_buffer(
-        filepath_or_buffer)
+    cdef cudf_io_types.source_info source
+    cdef const unsigned char[::1] buffer \
+        = lib.view_of_buffer(filepath_or_buffer)
     cdef string filepath
     if buffer is None:
-        if not os.path.isfile(filepath_or_buffer):
-            raise FileNotFoundError(
-                errno.ENOENT, os.strerror(errno.ENOENT), filepath_or_buffer
-            )
-        filepath = <string>str(filepath_or_buffer).encode()
-
-    cdef unique_ptr[parquet_reader] reader
-    with nogil:
-        if buffer is None:
-            reader = unique_ptr[parquet_reader](
-                new parquet_reader(filepath, options)
-            )
-        else:
-            reader = unique_ptr[parquet_reader](
-                new parquet_reader(<char*>&buffer[0], buffer.shape[0], options)
-            )
-
-    # Read data into columns
-    cdef cudf_table c_out_table
-    cdef size_type c_skip_rows = skip_rows if skip_rows is not None else 0
-    cdef size_type c_num_rows = num_rows if num_rows is not None else -1
-    cdef size_type c_row_group = row_group if row_group is not None else -1
-    with nogil:
-        if c_skip_rows != 0 or c_num_rows != -1:
-            c_out_table = reader.get().read_rows(c_skip_rows, c_num_rows)
-        elif c_row_group != -1:
-            c_out_table = reader.get().read_row_group(c_row_group)
+        if os.path.isfile(filepath_or_buffer):
+            filepath = <string>str(filepath_or_buffer).encode()
         else:
-            c_out_table = reader.get().read_all()
+            buffer = filepath_or_buffer.encode()
+
+    if buffer is None:
+        source.type = cudf_io_types.io_type.FILEPATH
+        source.filepath = filepath
+    else:
+        source.type = cudf_io_types.io_type.HOST_BUFFER
+        source.buffer.first = <char*>&buffer[0]
+        source.buffer.second = buffer.shape[0]
```

Comment:
Why not use [`make_source_info`](https://github.com/rapidsai/cudf/blob/3f0859824cae915e246a41be17d4d4789ec72b9f/python/cudf/cudf/_libxx/io/utils.pyx#L11) from `_libxx.io.utils` here?

```diff
+    cdef cudf_io_types.source_info source = make_source_info(filepath_or_buffer)
-    cdef cudf_io_types.source_info source
-    cdef const unsigned char[::1] buffer \
-        = lib.view_of_buffer(filepath_or_buffer)
-    cdef string filepath
-    if buffer is None:
-        if os.path.isfile(filepath_or_buffer):
-            filepath = <string>str(filepath_or_buffer).encode()
-        else:
-            buffer = filepath_or_buffer.encode()
-
-    if buffer is None:
-        source.type = cudf_io_types.io_type.FILEPATH
-        source.filepath = filepath
-    else:
-        source.type = cudf_io_types.io_type.HOST_BUFFER
-        source.buffer.first = <char*>&buffer[0]
-        source.buffer.second = buffer.shape[0]
```

Reply:
oh nice! Didn't know that existed and will gladly use that. Thanks!